homeassistant/components/amcrest/camera.py

```diff
@@ -132,7 +135,22 @@ async def async_setup_platform(
 
     name = discovery_info[CONF_NAME]
     device = hass.data[DATA_AMCREST][DEVICES][name]
-    async_add_entities([AmcrestCam(name, device, hass.data[DATA_FFMPEG])], True)
+    entity = AmcrestCam(name, device, hass.data[DATA_FFMPEG])
+
+    # 2021.9.0 introduced unique id's for the camera entity, but these were not
+    # unique for different resolution streams.  If any cameras were configured
+    # with this version, update the old entity with the new unique id.
```

Comment:
I'm not sure what is the policy for deprecating this migration logic.

Reply:
The only policy is to let it stay for a long time, because many do not upgrade every month. I just removed similar code in tradfri after 2 years.  In other words leave it until you have a good reason to remove it.